huaweicloud/services/hss/data_source_huaweicloud_hss_ransomware_protection_policies.go

```diff
@@ -0,0 +1,243 @@
+	var (
+		cfg         = meta.(*config.Config)
+		region      = cfg.GetRegion(d)
+		epsId       = cfg.DataGetEnterpriseProjectID(d)
+		limit       = int32(20)
+		offset      int32
+		allPolicies []hssv5model.ProtectionPolicyInfo
+	)
+
+	client, err := cfg.HcHssV5Client(region)
+	if err != nil {
+		return diag.Errorf("error creating HSS v5 client: %s", err)
+	}
+
+	for {
+		request := hssv5model.ListProtectionPolicyRequest{
+			Region:              region,
+			EnterpriseProjectId: utils.String(epsId),
+			Limit:               utils.Int32(limit),
+			Offset:              utils.Int32(offset),
+			ProtectPolicyId:     utils.StringIgnoreEmpty(d.Get("policy_id").(string)),
+			PolicyName:          utils.StringIgnoreEmpty(d.Get("name").(string)),
+			OperatingSystem:     utils.StringIgnoreEmpty(d.Get("operating_system").(string)),
+		}
+
+		listResp, listErr := client.ListProtectionPolicy(&request)
+		if listErr != nil {
+			return diag.Errorf("error querying HSS ransomware protection policies: %s", listErr)
+		}
+
+		if listResp == nil || listResp.DataList == nil {
+			break
+		}
+		if len(*listResp.DataList) == 0 {
+			break
+		}
+
+		allPolicies = append(allPolicies, *listResp.DataList...)
+		offset += limit
+	}
+
+	uuId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+
+	d.SetId(uuId)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("policies", flattenPolicies(filterPolicies(allPolicies, d))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func filterPolicies(policies []hssv5model.ProtectionPolicyInfo, d *schema.ResourceData) []hssv5model.ProtectionPolicyInfo {
+	if len(policies) == 0 {
+		return nil
+	}
+
+	protectionMode, ok := d.GetOk("protection_mode")
+	if !ok {
+		return policies
+	}
+
+	rst := make([]hssv5model.ProtectionPolicyInfo, 0, len(policies))
+	for _, v := range policies {
+		if fmt.Sprint(protectionMode) == utils.StringValue(v.ProtectionMode) {
+			rst = append(rst, v)
+		}
+	}
+
+	return rst
+}
+
+func flattenPolicies(policies []hssv5model.ProtectionPolicyInfo) []interface{} {
+	if len(policies) == 0 {
+		return nil
+	}
+
+	rst := make([]interface{}, 0, len(policies))
+	for _, v := range policies {
+		rst = append(rst, map[string]interface{}{
+			"id":                       v.PolicyId,
+			"name":                     v.PolicyName,
+			"protection_mode":          v.ProtectionMode,
+			"bait_protection_status":   v.BaitProtectionStatus,
+			"deploy_mode":              v.DeployMode,
+			"protection_directory":     v.ProtectionDirectory,
+			"protection_type":          v.ProtectionType,
+			"exclude_directory":        v.ExcludeDirectory,
+			"runtime_detection_status": v.RuntimeDetectionStatus,
+			"count_associated_server":  v.CountAssociatedServer,
+			"operating_system":         v.OperatingSystem,
+			"process_whitelist":        flattenProcessWhitelist(*v.ProcessWhitelist),
```

Comment:
Please precheck whether `v.ProcessWhitelist` is nil pointer.

Reply:
done.